script/deploy/deploySingleContract.sh

```diff
@@ -94,12 +94,18 @@ deploySingleContract() {
   # We need to use zksync specific scripts that are able to be compiled for
   # the zkvm
   if [[ $NETWORK == "zksync" ]]; then
+    # update the deploy script directory to point to zksync-specific scripts
     DEPLOY_SCRIPT_DIRECTORY="script/deploy/zksync/"
   fi
 
   if [[ -z "$CONTRACT" ]]; then
-    # get user-selected deploy script and contract from list
-    SCRIPT=$(ls -1 "$DEPLOY_SCRIPT_DIRECTORY" | sed -e 's/\.s.sol$//' | grep 'Deploy' | gum filter --placeholder "Deploy Script")
+    if [[ $NETWORK == "zksync" ]]; then
+      # get user-selected deploy script and contract from list
+      SCRIPT=$(ls -1 "$DEPLOY_SCRIPT_DIRECTORY" | sed -e 's/\.zksync.s.sol$//' | grep 'Deploy' | gum filter --placeholder "Deploy Script")
+    else
+      # get user-selected deploy script and contract from list
+      SCRIPT=$(ls -1 "$DEPLOY_SCRIPT_DIRECTORY" | sed -e 's/\.s.sol$//' | grep 'Deploy' | gum filter --placeholder "Deploy Script")
+    fi
```

Comment:
Consider avoiding redundancy by extracting only the file extension based on the network

Reply:
could you please elaborate on how you would do that in this case?
Every approach I can think of would add lines of code...

Reply:
@0xDEnYO pseudo code like:

`
EXTENSION=".s.sol"
if (NETWORK == "zksync") {
   EXTENSION=".zksync.s.sol"
}
`
 
 and then after it you need one line:
 
`
 SCRIPT=$(ls -1 "$DEPLOY_SCRIPT_DIRECTORY" | sed -e 's/\EXTENSION//' | grep 'Deploy' | gum filter --placeholder "Deploy Script")
`
 
 and in the lines: 125 to 129 you don't need to check network and instead you can have one line:
`
 FULL_SCRIPT_PATH="${DEPLOY_SCRIPT_DIRECTORY}${SCRIPT}${EXTENSION}"
`

Reply:
Thanks a lot for the clarification. 
Initially I was hesitant cause it would just add lines and is more difficult to read but because we can save some lines of code further down the line, I really like the suggestion and have updated it accordingly :)